starfish/image/_stack.py

```diff
@@ -28,15 +28,32 @@ def __init__(self, image_partition):
             self._num_zlayers = 1
         self._tile_shape = tuple(image_partition.default_tile_shape)
 
-        self._data = numpy.zeros((self._num_hybs, self._num_chs, self._num_zlayers) + self._tile_shape)
-        self._data_needs_writeback = False
+        kind = None
```

Comment:
I added a small snippet of a comment for explaining what is happening here.

The requirement I'm imposing here is that they have the same kind but I'm not imposing that they have the same size.  I allocate the largest size I see.  Theoretically, that would mean we should be able to handle all the tiles with smaller data sizes.  We would not clip because we would then write out the largest data size.

I am open, however, to just banning mismatched data types.

Reply:
My experience leads me to believe that we should ban mismatched data types. However, I'd like a second opinion here as I'm relatively new to the data. @freeman-lab Is there enough information here for you to weigh in? 

One more piece of background: 
`kind` = the type of stored data, e.g. unsigned integer (uint32, uint16, uint8 are all same kind, but int32 would be a different kind
`size` = number of bits (8, 16, 32, above)
`dtype` = kind + size. (e.g. uint + 32)

I'm asserting all tiles need the same dtype, not just the same kind, because otherwise dynamic ranges + interpretation get all funky. I'm not confident enough to make that decision unilaterally, though.